src/hostapi/pulseaudio/pa_linux_pulseaudio.c

```diff
@@ -67,50 +67,65 @@ extern char *__progname;
 /* PulseAudio specific functions */
 int PaPulseAudio_CheckConnection( PaPulseAudio_HostApiRepresentation * ptr )
 {
-    pa_context_state_t state;
+    int retCode = paNotInitialized;
 
-
-    /* Sanity check if ptr if NULL don't go anywhere or
-     * it will SIGSEGV
-     */
-    if( !ptr )
+    if ( ptr == NULL )
     {
-        PA_PULSEAUDIO_SET_LAST_HOST_ERROR( 0,
-                                           "PaPulseAudio_CheckConnection: Host API is NULL! Can't do anything about it" );
-        return -1;
+        return retCode;
     }
 
-    if( !ptr->context || !ptr->mainloop )
+    pa_context_state_t state = pa_context_get_state( ptr->context );
+
+    /* Check if contenxt if good.
+     * If it is then check state
+     */
+    if( PA_CONTEXT_IS_GOOD(state) )
     {
-        PA_PULSEAUDIO_SET_LAST_HOST_ERROR( 0,
-                                           "PaPulseAudio_CheckConnection: PulseAudio context or mainloop are NULL" );
-        return -1;
-    }
+        switch( state )
+        {
+            /* These can be found from
+             * https://freedesktop.org/software/pulseaudio/doxygen/def_8h.html
+             */
 
-    state = pa_context_get_state(ptr->context);
+            case PA_CONTEXT_READY:
+                retCode = paNoError;
+            break;
 
-    if( !PA_CONTEXT_IS_GOOD(state) )
+            case PA_CONTEXT_CONNECTING:
+            case PA_CONTEXT_AUTHORIZING:
+            case PA_CONTEXT_SETTING_NAME:
+                /* These are just here if they are needed in future */
+            break;
+        }
+    }
+    else
     {
+        retCode = paUnanticipatedHostError;
```

Comment:
```suggestion
```

Just delete this line and continue with `paNotInitialized` from above. You can't return `paUnanticipatedHostError;` without setting the host error.